app/controllers/revisions_controller.rb

```diff
@@ -57,12 +57,43 @@ def latest_diff
     end
   end
 
+  # PATCH /notebooks/:notebook_id/revisions/:commit_id/edit_friendly_label
+  def edit_friendly_label
+    errors = ""
+    notebook = Notebook.find(@revision.notebook_id)
+    friendly_label = params[:friendly_label].strip
+    if verify_revision_label(friendly_label, notebook, params[:old_label])
+      errors += verify_revision_label(friendly_label, notebook, params[:old_label])
+    end
+    if errors.length <= 0
+      if friendly_label.strip != ""
+        @revision.friendly_label = friendly_label
+      else
+        @revision.friendly_label = nil
+      end
+      @revision.save!
+      # Refresh page if user resets the version, else inline alert it was updated successfully
+      if @revision.friendly_label == nil
+        flash[:success] = 'Friendly label for revision has been reset successfully.'
+        if request.xhr?
+          render :js => %(window.location.href='#{notebook_revisions_path(@notebook.id)}')
+        else
+          redirect_back(fallback_location: root_path)
+        end
+      else
+        render json: { message: 'Friendly label for revision has been updated successfully.' }, status: :ok
+      end
+    else
+      render json: { message: errors }, status: :unprocessable_entity
+    end
+  end
+
   # PATCH /notebooks/:notebook_id/revisions/:commit_id/edit_summary
   def edit_summary
     errors = ""
     revision_summary = params[:summary].strip
     if revision_summary.length > 250
-      errors += "Revision summary was too long. Only accepts 250 characters and you submitted one that was #{revision_summary.length} characters."
+      errors += "Revision summary was too long. Only accepts a maximum of 250 characters and you submitted one that was #{revision_summary.length} characters. "
```

Comment:
"Only accepts...." This still feels like clumsy grammar even with the edit
"Summary can be a maximum of 250 characters..." ?

Reply:
Fixed